py/sight/widgets/decision/decision.py

```diff
@@ -317,398 +314,384 @@ def get_decision_messages_from_proto(
+
+  #               if env:
+  #                 driver_fn(env, sight)
+  #               else:
+  #                 driver_fn(sight)
+
+  #               finalize_episode(sight)
+  #               sight.exit_block('Decision Sample', sight_pb2.Object())
+  #             else:
+  #               raise ValueError("invalid response from server")
+  #           logging.info('exiting from the loop.....')
+  #         # else:
+  #         #   for _ in range(num_samples_to_run):
+  #         #     sight.enter_block('Decision Sample', sight_pb2.Object())
+  #         #     if 'constant_action' in sight.widget_decision_state:
+  #         #         del sight.widget_decision_state['constant_action']
+  #         #     sight.widget_decision_state['discount'] = 0
+  #         #     sight.widget_decision_state['last_reward'] = None
+
+  #         #     if env:
+  #         #         driver_fn(env, sight)
+  #         #     else:
+  #         #         driver_fn(sight)
+
+  #         #     finalize_episode(sight)
+  #         #     sight.exit_block('Decision Sample', sight_pb2.Object())
+
+  #         # req = service_pb2.TestRequest(client_id=str(sight.id))
+  #         # response = service.call(
+  #         #     lambda s, meta: s.PrintInsertionTime(req, 300, metadata=meta)
+  #         # )
+
+  #       logging.debug("<<<<  Out %s of %s", method_name, _file_name)
+  # #! need to discard this condition when yaml flow is setup
+  # `else:`
 
   sight.widget_decision_state['num_decision_points'] = 0
-  sight.widget_decision_state['proposed_actions'] = []
-
-  decision_mode_actions = {
-      'run': execute_run_mode,
-      'configured_run': (lambda sight=sight, driver_fn=driver_fn:
-                         execute_configured_run_mode(sight, driver_fn)),
-      'train':
-          (lambda sight=sight, decision_configuration=decision_configuration,
-           driver_fn=driver_fn, env=env: execute_train_mode(
-               sight, decision_configuration, driver_fn, env)),
-  }
 
-  action = decision_mode_actions.get(_DECISON_MODE.value)
-  if action:
-    action()
+  if configs:
+    # *** This configs are for each question's meta data
+    question_config = configs.get('question_config', {})
+    optimizer_config = configs.get('optimizer_config', {})
+    optimizer_type = optimizer_config.get('optimizer', None)
+    workers_config = configs.get('workers_config', {})
+    question_label = question_label
+    optimizer.obj = setup_optimizer(sight, optimizer_type)
+    decision_configuration = configure_decision(sight, question_label,
+                                                question_config,
+                                                optimizer_config, optimizer.obj)
+
+    ## ! Don't know why we did this ?
+    sight.widget_decision_state['proposed_actions'] = []
+
+    decision_mode_actions = {
+        'run': execute_run_mode,
+        'configured_run': (lambda sight=sight, driver_fn=driver_fn:
```

Comment:
pull these lambdas into regular functions. This is too dense to be readable.

In fact, do you actually need them? it seems like you can just pass a reference to functions execute_configured_run_mode and execute_train_mode directly with no lambda